src/ray/core_worker/task_manager.h

```diff
@@ -142,19 +142,24 @@ class TaskManager : public TaskFinisherInterface, public TaskResubmissionInterfa
   /// \param[in] creation_task_exception If this arg is set, it means this task failed
   /// because the callee actor is dead caused by an exception thrown in creation task,
   /// only applies when error_type=ACTOR_DIED.
-  /// \param[in] immediately_mark_object_fail whether immediately mark the task
-  /// result object as failed.
+  /// \param[in] mark_task_object_failed whether or not it marks the task
+  /// return object as failed.
   /// \return Whether the task will be retried or not.
-  bool PendingTaskFailed(const TaskID &task_id, rpc::ErrorType error_type,
-                         const Status *status = nullptr,
-                         const rpc::RayException *creation_task_exception = nullptr,
-                         bool immediately_mark_object_fail = true) override;
-
-  /// Treat a pending task as failed. The lock should not be held when calling
-  /// this method because it may trigger callbacks in this or other classes.
-  void MarkPendingTaskFailed(const TaskSpecification &spec, rpc::ErrorType error_type,
-                             const rpc::RayException *creation_task_exception =
-                                 nullptr) override LOCKS_EXCLUDED(mu_);
+  bool FailOrRetryPendingTask(const TaskID &task_id, rpc::ErrorType error_type,
```

Comment:
RetryUntilExhausted?

Reply:
Is it a strongly preferred name? Kind of prefer to have "fail" in the method name (since that's one of features of this API), so I'd like to keep it if you don't have strong preference the new name (but I can change it otherwise).